### PR TITLE
MIT Professional Education course import

### LINKED ISSUE #2737: Add MIT Professional course offerings
https://professional.mit.edu -> offered by "Professional Education"

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Closes #2737

#### What's this PR do?
- Imports courses from https://professional.mit.edu
- Maps topics for the above courses (and Sloan courses) to EdX topics

#### How should this be manually tested?
- Add `MITPE_BASE_URL=https://professional.mit.edu/` to your .env file
- Run `manage.py backpopulate_mitpe_data`
- Go to the course search page, there should be about 80 Professional Education courses there

#### Screenshots
Screen Shot 2020-04-06 at 3 09 23 PM

Screen Shot 2020-04-06 at 3 09 42 PM

### CODE REVIEW

**Comment by mbertrand** [comment]
@pdpinch @abdulkdawson @Ferdi screenshots are in the description, the current logo is from a screenshot I took of the website.

**Comment by pdpinch** [comment]
Did the Subject area list expand? I don't remember "Crisis Management" or "Radar" being there.

**Comment by mbertrand** [comment]
@pdpinch yes, the subjects come from the course listings, and will be created if they don't already exist.

File: course_catalog/etl/mitpe.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    log_exceptions,
+    generate_unique_id,
+    strip_extra_whitespace,
+    parse_dates,
+    map_topics,
+)
+
+log = logging.getLogger()
+
+OFFERED_BY = [{"name": OfferedBy.mitpe.value}]
```
Comment (abeglova) [inline course_catalog/etl/mitpe.py:23]: I don't think this is used

File: course_catalog/etl/mitpe.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    )
+    listings = soup.find("div", {"class": "course-listing"}).findAll(
+        "div", {"class": "views-row"}
+    )
+    for listing in listings:
+        link = listing.find("a")
+        url = urljoin(settings.MITPE_BASE_URL, link.get("href"))
+        details = bs(requests.get(url).content, "html.parser")
+        description = _parse_description(details)
+        short_description = strip_extra_whitespace(description[0])
```
Comment (abeglova) [inline course_catalog/etl/mitpe.py:154]: Some of the first paragraphs are a bit too general to be a good course description, but I don't have a better solution

File: course_catalog/etl/mitpe.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            "offered_by": [{"name": OfferedBy.mitpe.value}],
+            "runs": [
+                {
+                    "url": course["url"],
+                    "prices": ([{"price": course["price"]}] if course["price"] else []),
+                    "run_id": generate_unique_id(
+                        f"{course['url']}{datetime.strftime(date_range[0], '%Y%m%d')}"
+                    ),
+                    "platform": PlatformType.mitpe.value,
+                    "start_date": date_range[0],
```
Comment (abeglova) [inline course_catalog/etl/mitpe.py:195]: We probably don't care but the day before the start date is displayed as the start date in the drawer due to UTC conversion issues

Comment (pdpinch) [reply course_catalog/etl/mitpe.py:195]: I care. Can we set the start time to something arbitrary, but likely, like 9am EST ?

Comment (mbertrand) [reply course_catalog/etl/mitpe.py:196]: Good catch.  Changed to a default of noon UTC (== 7-8 Eastern, 4-5 Pacific)

File: course_catalog/etl/mitpe.py (hunk 1 above)
Comment (mbertrand) [reply course_catalog/etl/mitpe.py:23]: used now in transform

File: course_catalog/etl/mitpe.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                    "offered_by": [{"name": OfferedBy.mitpe.value}],
+                    "title": course["title"],
+                    "short_description": course["short_description"],
+                    "full_description": course["full_description"],
+                    "instructors": [
+                        {"first_name": first_name, "last_name": last_name}
+                        for (first_name, last_name) in course["instructors"]
+                    ],
+                }
+                for date_range in course["dates"]
```
[the remark this answers is not in the document]
Comment (mbertrand) [reply course_catalog/etl/mitpe.py:209]: skipping courses without runs now

**Comment by mbertrand** [comment]
@abeglova ready for another look

File: course_catalog/etl/mitpe.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (abeglova) [reply course_catalog/etl/mitpe.py:209]: The script does not unpublish courses that have already been added to the search. Do we need it to?

[the remark this answers is not in the document]
Comment (mbertrand) [reply course_catalog/etl/mitpe.py:209]: I don't think so

[the remark this answers is not in the document]
Comment (mbertrand) [reply course_catalog/etl/mitpe.py:209]: @pdpinch should old Sloan/Professional/CSAIL courses without any current/upcoming runs still be searchable or not?  EdX courses are, for example - https://discussions-rc.odl.mit.edu/learn/?lr_id=8700&type=course

[the remark this answers is not in the document]
Comment (mbertrand) [reply course_catalog/etl/mitpe.py:209]: Ferdi says it's okay as is for now

**Pull request merged**